ctms/ingest_stripe.py

```diff
@@ -350,15 +371,6 @@ def ingest_stripe_invoice(
         invoice.status = data["status"]
         invoice.default_payment_method_id = data["default_payment_method"]
         invoice.default_source_id = data["default_source"]
-        lines_to_delete = {
-            line_id
-            for line_id, in (
-                db_session.query(StripeInvoiceLineItem.stripe_id)
-                .with_for_update()
-                .filter(StripeInvoiceLineItem.stripe_invoice_id == invoice.stripe_id)
-                .all()
```

Comment:
Not only did this query fetch line items before we used them, but it also fetched more than we needed, since it fetches all of the line items, then later we remove line items from the set that appear in the incoming invoice data.